src/OpenTelemetry.Api/Context/Propagation/AWSXRayPropagator.cs

```diff
@@ -0,0 +1,276 @@
+// <copyright file="AWSXRayPropagator.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Numerics;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Context.Propagation
+{
+    /// <summary>
+    /// Propagator for AWS X-Ray. See https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-tracingheader.
+    /// </summary>
+    public class AWSXRayPropagator : IPropagator
+    {
+        private const string AWSXRayTraceHeaderKey = "X-Amzn-Trace-Id";
+        private const char KeyValueDelimiter = '=';
+        private const char TraceHeaderDelimiter = ';';
+
+        private const string RootKey = "Root";
+        private const int Version = 1;
+        private const int RandomNumberHexDigits = 24;
+        private const int EpochHexDigits = 8;
+        private const int VersionDigits = 1;
+        private const int ElementsCount = 3;
+        private const int TotalLength = RandomNumberHexDigits + EpochHexDigits + VersionDigits + ElementsCount - 1;
+        private const char TraceIdDelimiter = '-';
+
+        private const string ParentKey = "Parent";
+        private const int ParentIdHexDigits = 16;
+
+        private const string SampledKey = "Sampled";
+        private const char SampledValue = '1';
+        private const char NotSampledValue = '0';
+
+        private static readonly char[] ValidSeparators = { ';' };
+        private static readonly HashSet<string> AllFields = new HashSet<string>() { AWSXRayTraceHeaderKey };
```

Comment:
Do you need to make this immutable since you return it to users?

Reply:
Will update in v2